packages/providers/src.ts/base-provider.ts

```diff
@@ -760,14 +761,24 @@ export class BaseProvider extends Provider {
             if (data) {
                 const filtered = data.transactions.filter((e: { result: string; }) => e.result != 'DUPLICATE_TRANSACTION');
                 if (filtered.length > 0) {
-                    const contractsEndpoint = MIRROR_NODE_CONTRACTS_ENDPOINT + transactionId;
-                    const dataWithLogs = await axios.get(this._mirrorNodeUrl + contractsEndpoint);
-                    const record = {
+                    let record: any;
+                    record = {
                         chainId: this._network.chainId,
                         transactionId: transactionId,
                         result: filtered[0].result,
-                        ...dataWithLogs.data
                     };
+
+                    const transactionName = filtered[0].name;
+                    if (transactionName === 'CRYPTOCREATEACCOUNT') {
+                        record.hash = filtered[0].transaction_hash;
```

Comment:
For Contract related TX we are using different format for transaction hash. Instead of having the base64 encoded version for crypto creates and the hex encoded (keccak) version for Contract related transactions, lets compute the hex encoded version for `CRYPTOCREATEs` and populate it in the `hash` property.

Reply:
Done, but I'm not sure if it's correct. Will ask a few questions in next meeting.

Reply:
Applied the correct hash transforming logic.